File: 준영/LV2/다음 큰 숫자.py
def solution(n):
    answer = 0

    result = []
    temp = n
    while temp:
        result.append(temp%2)
        temp //= 2

    idx = 0
    while True:
        idx += 1
        arr = []
        temp = n + idx
        while temp:
            arr.append(temp%2)
            temp //= 2

        if arr.count(1) == result.count(1):
            answer = n + idx
            break

    return answer

File: 준영/프로그래머스/LV2/다음 큰 숫자.js
function solution(n) {
  var answer = 0
  var result = []
  var temp = n

  while (temp) {
    result.push(temp%2)
    temp = parseInt(temp/2)
  }

  const cnt = result.filter((x) => x === 1).length

  var idx = 0
  while (true) {
    idx += 1
    var arr = Array(0)
    temp = n + idx
    while (temp) {
      arr.push(temp%2)
      temp = parseInt(temp/2)
    }

    if (cnt === arr.filter((x) => x === 1).length) {
      answer = n + idx
      break
    }

  }
  return answer
}

solution(78)